src/mpi/errhan/dynerrutil.c

```diff
@@ -413,7 +416,7 @@ int MPIR_Remove_error_code_impl(int code)
 
     struct intcnt *s;
 
-    if (class & ERROR_DYN_MASK) {
+    if (code & ERROR_DYN_CLASS) {
```

Comment:
Hm, OK. I think I sort of understand what's happening now. Did the previous version of this branch just not work? Or is this change equivalent?

Reply:
The previous version used the highest bit, which makes the error code negative in value. That beaks the usage of comparing against `MPI_ERR_LASTCODE`.

Reply:
`ERROR_DYN_MASK` is `0x40000000` so not the sign bit as far as I can tell? I'm still not sure why `class & ERROR_DYN_MASK` needs to change here.